src/core/exec-invoke.c

```diff
@@ -2251,6 +2252,126 @@ static int setup_private_users(PrivateUsers private_users, uid_t ouid, gid_t ogi
         return 1;
 }
 
+static int can_mount_proc(const ExecContext *c, ExecParameters *p) {
+        _cleanup_close_pair_ int errno_pipe[2] = EBADF_PAIR;
+        _cleanup_(sigkill_waitp) pid_t pid = 0;
+        _cleanup_(rmdir_and_freep) char *tmpdir = NULL;
+        ssize_t n;
+        int r;
+
+        assert(c);
+        assert(p);
+
+        /* If running via unprivileged user manager and /proc/ is masked (e.g. /proc/kmsg is over-mounted with tmpfs
+         * like systemd-nspawn does), then mounting /proc/ will fail with EPERM. This is due to a kernel restriction
+         * where unprivileged user namespaces cannot mount a less restrictive instance of /proc. */
+
+        /* Create temporary directory for mounting /proc. */
+        r = mkdtemp_malloc("/tmp/systemd-temporary-proc-XXXXXX", &tmpdir);
```

Comment:
Is this dir really needed? Can probably use some known existing dir, e.g. /dev/shm/? (Our credential logic uses that in the intermediate mount ns too)

Reply:
/dev/shm works for me!